### PR TITLE
[CFX-4964] Create/Edit drconfig.yaml only after valid API key received

### PR DESCRIPTION
**Pull request by taras-pokornyy**
# RATIONALE

Don't touch ~/.config/datarobot/drconfig.yaml until both valid endpoint AND token are ready when running dr auth login. So unfinished auth flow won't break anything.

Actual:

When running dr auth login, the CLI will write to ~/.config/datarobot/drconfig.yaml a set of credentials before the auth flow is complete.

Expected:

When running dr auth login, the CLI will write to ~/.config/datarobot/drconfig.yaml the set of credentials ONLY once auth flow is complete and successful.

## CHANGES
- Created `SetURLToConfig` function in `internal/config/api.go`
- Updated following files to use new `SetURLToConfig` function instead of `SaveURLToConfig`:
-- `cmd/auth/login/cmd.go`
-- `cmd/auth/seturl/cmd.go`
-- `internal/auth/auth.go`

## IMPORTANT
`SaveURLToConfig` function still exists as it is used in `cmd/templates/setup/model.go`

## PR Automation

**Comment-Commands:** Trigger CI by commenting on the PR:
- `/trigger-smoke-test` or `/trigger-test-smoke` - Run smoke tests
- `/trigger-install-test` or `/trigger-test-install` - Run installation tests

**Labels:** Apply labels to trigger workflows:
- `run-smoke-tests` or `go` - Run smoke tests on demand (only works for non-forked PRs)

> [!IMPORTANT]
> **For Forked PRs:** If you're an external contributor, the `run-smoke-tests` label won't work. A maintainer must manually trigger the "Fork PR Smoke Tests" workflow from the Actions tab, providing your PR number. Please comment requesting a maintainer review if you need smoke tests to run.

---

> [!NOTE]
> **Medium Risk**
> Changes when the CLI mutates and writes `drconfig.yaml` during authentication, which can affect users’ ability to log in and persist credentials if there are edge-case failures. The scope is contained to URL-setting and config-write helpers, with added tests to reduce regression risk.
> 
> **Overview**
> **Defers config-file writes during auth URL setup.** `dr auth login` and `dr auth set-url` now call the new `config.SetURLToConfig` helper, which updates the in-memory Viper config with a normalized endpoint (`/api/v2`) **without writing** `drconfig.yaml` until the auth flow completes successfully.
> 
> `auth.WriteConfigFileSilent` now ensures the config directory/file exists before writing, and the interactive URL prompt messaging is updated to reflect that validation/API-key retrieval happens next. Adds a new `internal/config/api_test.go` suite covering URL normalization (including shortcuts) and asserting `SetURLToConfig` does not write to disk.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 69035964918543aa3bb8ccd47c0f285580c2bdfd. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by ajalon1 — CHANGES_REQUESTED** [review]
Requesting changes to make sure that we handle all scenarios where we write to config. Good first pass.

I'd also suggest writing unit tests for SetUrlToConfig(), and ensure that the config file is NOT written to.

File: internal/auth/auth.go [hunk 1]
```diff
@@ -256,6 +256,10 @@ func WaitForAPIKeyCallback(ctx context.Context, datarobotHost string) (string, e
 }
 
 func WriteConfigFileSilent() error {
+	// Ensure the config directory and file exist before writing the config file
+	if err := config.CreateConfigFileDirIfNotExists(); err != nil {
+		return err
+	}
 	err := viper.WriteConfig()
```
[the remark this answers is not in the document]
Comment (taras-pokornyy) [reply internal/auth/auth.go]: I have some issues running linter locally. Have to fix them 😅 
Thank you for pointing on that

File: internal/config/api.go [hunk 2]
```diff
@@ -82,6 +82,8 @@ func RedactedReqInfo(req *http.Request) string {
 	return string(requestDump)
 }
 
+// TODO: I believe we want to delete this function as there is SetURLToConfig function
+// But it is used in cmd/templates/setup/model.go
```
[the remark this answers is not in the document]
Comment (ajalon1) [reply internal/config/api.go:88]: LOL, I don't think they will have an issue. I see @victorborshak fixed those in #416 last week :)

File: internal/config/api.go [hunk 3]
```diff
@@ -106,6 +108,20 @@ func SaveURLToConfig(newURL string) error {
...
+// SetURLToConfig is a helper function that sets the DataRobot URL with the "/api/v2" suffix in the config object.
+// It is used by both cmd/auth/set-url and cmd/auth/login to ensure consistent URL formatting.
+// It does NOT write to the config file, in order not to break drconfig.yaml file once URL is not valid or some issues with API key.
+func SetURLToConfig(newURL string) error {
+	newURL, err := SchemeHostOnly(urlFromShortcut(newURL))
+	if err != nil {
+		return err
+	}
+
+	viper.Set(DataRobotURL, newURL+"/api/v2")
```
[the remark this answers is not in the document]
Comment (taras-pokornyy) [reply internal/config/api.go]: Done! 
```const DRAPIURLSuffix = "/api/v2"```

**Review by c-h-russell-walker — APPROVED** [review]

**Review by ajalon1 — APPROVED** [review]
🚢
